origin-js/src/models/listing.js

```diff
@@ -3,17 +3,120 @@
 //
 export class Listing {
   /**
-   * A Listing is constructed based on its on-chain and off-chain data.
-   * @param {string} listingId - Unique listing ID.
-   * @param {Object} chainListing - Listing data from the blockchain.
-   * @param {Object} ipfsListing - Listing data from IPFS.
+   * Listing object model.
+   *
+   * @param {Object} args - single object arguments used to construct a Listing
+   *  - {string} id
+   *  - {string} title
+   *  - {string} description
+   *  - {string} category
+   *  - {Object} commission - consists of 'amount' and 'currency' properties
```

Comment:
nit: could be document in JSDoc style this way I think:
{{amount: string, currency: string}}
Same comment for all the other prices below.